### PR TITLE
Migration Babel

### PR DESCRIPTION
**Pull request by DevSide**
Modernize repository

- [x] build under lib/ (commonjs) and es/ with babel
- [x] introduce `const` and `let`
- [x] update eslint rules
- [x] yarn

### CODE REVIEW

**Comment by flepretre** [comment]
> migration nodeunit => jest

It should be in an other PR

File: .gitignore [hunk 1]
```diff
@@ -1 +1,5 @@
 node_modules
+.DS_Store
```
Comment (flepretre) [inline .gitignore]: `.DS_Store` is for mac users only, isn't it ? It should not be here.

File: README.md [hunk 2]
```diff
@@ -25,11 +25,9 @@ See [this post](http://tech.m6web.fr/how-did-we-mock-the-backend-developers.html
 
 ## Installation
 
-Install with [npm](http://npmjs.org/):
+Install with [npm](http://npmjs.org/): `npm install superagent-mock`
```
Comment (flepretre) [inline README.md]: Do we still want to support npm ?

File: tests/support/expectations.js [hunk 3]
```diff
@@ -10,826 +10,844 @@ module.exports = function (request, config, isServer) {
...
-      // Stub setTimeout
-      Object.defineProperty(global, 'setTimeout', {
-        value: function(callbackFunc, timeout) {
-          if (!global.setTimeout.calls) {
-            global.setTimeout.calls = [];
-          }
-          global.setTimeout.calls.push([callbackFunc, timeout]); // spy on calls made to setTimeout
-          callbackFunc();
+    // Stub setTimeout
+    Object.defineProperty(global, 'setTimeout', {
```
Comment (flepretre) [inline tests/support/expectations.js]: Jest has its own way to deals with timeouts, we should use it.

File: tests/support/expectations.js [hunk 4]
```diff
@@ -10,826 +10,844 @@ module.exports = function (request, config, isServer) {
     currentLog = log;
   };
   var superagentPackage = require('superagent/package.json');
-  var superagentUserAgentHeader = isServer ? {"User-Agent": 'node-superagent/' + superagentPackage.version} : {};
+  var superagentUserAgentHeader = isServer ? {'User-Agent': 'node-superagent/' + superagentPackage.version} : {};
```
Comment (flepretre) [inline tests/support/expectations.js]: The test suite should be a describe block, not a module export.

File: README.md (hunk 2 above)
Comment (DevSide) [inline README.md]: yes !

File: .gitignore (hunk 1 above)
Comment (DevSide) [inline .gitignore]: Finally, it's already excluded by on `npm publish` https://docs.npmjs.com/misc/developers

File: tests/support/expectations.js (hunk 4 above)
Comment (DevSide) [inline tests/support/expectations.js]: We use the same test file for client and server side, it is convenient to have a function bulider here.

File: README.md (hunk 2 above)
Comment (D34THWINGS) [inline README.md]: Npm is the default package manager and so we need this

File: package.json [hunk 5]
```diff
@@ -15,15 +15,25 @@
     "url": "https://github.com/M6Web/superagent-mock"
   },
   "peerDependencies": {
-    "superagent": "^3.4.4"
+    "superagent": "^3.5.2"
```
Comment (D34THWINGS) [inline package.json]: Maybe it should be better not to use the `^` ?

File: package.json [hunk 6]
```diff
@@ -15,15 +15,25 @@
...
+    "cross-env": "^4.0.0",
+    "eslint": "^3.19.0",
+    "jest": "^19.0.2",
+    "rimraf": "^2.6.1",
+    "superagent": "^3.5.2"
   },
   "scripts": {
-    "test": "nodeunit ./tests",
-    "lint": "eslint ./lib"
+    "clean": "rimraf lib es",
```
Comment (D34THWINGS) [inline package.json]: rm -rf doesn't work here ? We could spare a dependency

File: package.json [hunk 7]
```diff
@@ -15,15 +15,25 @@
...
+    "test": "cross-env BABEL_ENV=commonjs jest --config .jestrc",
+    "lint": "eslint ./src ./tests",
+    "build:commonjs": "cross-env BABEL_ENV=commonjs babel src --out-dir lib",
+    "build:es": "cross-env BABEL_ENV=es babel src --out-dir es",
+    "build": "npm run build:commonjs && npm run build:es",
+    "prepublish": "yarn run clean && yarn run lint && yarn run build"
```
Comment (D34THWINGS) [inline package.json]: The `run` keyword after each `yarn` is useless

File: package.json [hunk 8]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "superagent-mock",
   "description": "superagent plugin allowing to simulate HTTP calls by returning data fixtures based on the requested URL.",
-  "version": "3.3.0",
+  "version": "3.4.0",
```
Comment (D34THWINGS) [inline package.json]: This should be done in a release commit I think, not here

File: tests/client.spec.js [hunk 9]
```diff
@@ -3,11 +3,11 @@
 global.window = {};
 
 // Get the "client" version of superagent
-var request = require("superagent/lib/client.js");
+const request = require('superagent/lib/client.js');
```
Comment (D34THWINGS) [inline tests/client.spec.js]: Remove `.js` extension

File: tests/support/config.js [hunk 10]
```diff
@@ -93,7 +93,7 @@ module.exports = [
   {
     pattern: 'https://error.example/(\\w+)',
     fixtures: function (match) {
-      var code = (match || [])[1] || 404;
+      var code = parseInt((match || [])[1]) || 404;
```
Comment (D34THWINGS) [inline tests/support/config.js]: Set radix param of `parseInt` to 10 or use the `Number()` constructor

File: package.json (hunk 8 above)
Comment (DevSide) [inline package.json]: I agree, or an other PR !

File: package.json (hunk 5 above)
Comment (DevSide) [inline package.json]: Not sure we want a strict peer dependency ?

File: package.json (hunk 6 above)
Comment (DevSide) [inline package.json]: `rimraf` like `cross-env`can be handle by Windows plateform :laughing:

File: package.json (hunk 7 above)
Comment (DevSide) [inline package.json]: You're right. Except for `clean` which is a yarn command

File: tests/support/config.js (hunk 10 above)
Comment (DevSide) [inline tests/support/config.js]: Number

File: package.json (hunk 6 above)
Comment (D34THWINGS) [inline package.json]: Ah yeah, I forgot that

File: package.json (hunk 5 above)
Comment (D34THWINGS) [inline package.json]: Oops, yeah not for PD, but we can do this for devDependencies maybe ?

Comment (DevSide) [inline package.json]: Every devDeps are starting with "^", so i prefer keep it for this PR

**Comment by DevSide** [comment]
@flepretre could you approuve please ?

**Comment by DevSide** [comment]
I've got a problem yarn/npm publish use the .gitignore so lib/ and es/folder are not published..

**Pull request merged**